### DISCUSSION TITLE
Set default group for signups

### DISCUSSION THREAD

**Discussion by wigman** [outline/outline#1662] [Self Hosting] [answered]
What would be the best way to automatically put people in a certain group?

I want to restrict write access to docs, so I've added permissions to all collections for an `author`, and a `reader` group.
By default, I'd like new signups to be able to only read all content.

Any ideas would be highly appreciated!

**Comment by tommoor** [comment]
This is a good idea, the ability to set one or more "default" groups – but it's not something that exists right now. If there is interest in adding this feature I could write up some guidance on how best to implement…

**Comment by wigman** [comment, marked as answer]
Thanks for your reply!

For the meantime, do you recon this would work? I'm no SQL expert and especially not PostgreSQL so would appreciate a second opinion :)

```SQL
CREATE OR REPLACE FUNCTION add_new_user_to_group()
  RETURNS trigger AS
$$
BEGIN
    INSERT INTO group_users
        ("userId",
        "groupId",
        "createdById",
        "createdAt",
        "updatedAt",
        "deletedAt")
    VALUES
        (NEW.id,
        (SELECT id from groups where name='readers'),
        (SELECT id from users where "isAdmin"='true'),
        NOW(),
        NOW(),
        null);
	
    RETURN NEW;
END
$$
  LANGUAGE plpgsql;


CREATE TRIGGER after_user_insert
    AFTER INSERT ON users
    FOR EACH ROW
    EXECUTE FUNCTION add_new_user_to_group();
```

**Reply by tommoor** [reply]
I'm also no sql expert but I think the general idea here would work well!

**Reply by wigman** [reply]
Thanks! Just tested it and it seems to work indeed. This'll be good enough for now 👍 
Thanks a lot for your time and for building a great product!

**Comment by matbrgz** [comment]
This could be a feature in Outline. In Settings, you can put users into a default group, what u think tom?

**Reply by tommoor** [reply]
Yep, totally agree – I think a control on the groups menu that lets you optionally select one as default
